### ISSUE TITLE
Becoming #PuppetApproved

### ISSUE THREAD

**Issue by jfryman** [voxpupuli/puppet-nginx#465]
The folks at PuppetLabs just released a set of guidelines to become #PuppetApproved. This issue will track progress toward that goal.

Full spec can be found here: https://forge.puppetlabs.com/approved/criteria
## Style
- [x] must not produce warnings in puppet-lint when run with this configuration:

``` ruby
PuppetLint.configuration.fail_on_warnings
PuppetLint.configuration.send('relative')
PuppetLint.configuration.send('disable_80chars')
PuppetLint.configuration.send('disable_class_inherits_from_params_class')
PuppetLint.configuration.send('disable_class_parameter_defaults')
PuppetLint.configuration.send('disable_documentation')
PuppetLint.configuration.send('disable_single_quote_string_with_variables')
PuppetLint.configuration.ignore_paths = ["spec//*.pp", "pkg//*.pp"]
```
## Documentation
- [x] must have a README
- [ ] README should conform to documentation standards (https://github.com/jfryman/puppet-nginx/pull/451)
- [x] must document example usage in README
- [ ] should completely document classes, defines, parameters, and resources used in the example in the README
## Maintenance & Lifecycle
- [x] should be regularly maintained and have received updates in the last 6 months
- [x] should not have more than 1 month gap between Forge and VCS
- [x] should be contributed to by more than one person or organization
## License
- [x] must be licensed
- [x] should be licensed under Apache, MIT, or BSD
## Metadata
- [x] must accurately express every required metadata field.
- [x] must express compatibility metadata for Puppet/PE and OS versions module is compatible with. (#598)
- [x] must accurately express issues_url and project_page metadata.
- [x] should provide accurate information for every metadata field
## SemVer
- [x] must be versioned according to SemVer v1 rules
- [x] must have candidate release >=1.x
## Testing
- [x] Should have rspec-puppet tests for manifests
- [x] Should have acceptance tests, preferably written with the beaker framework
- [x] Should have at least 1 unit test for each type, provider, fact, and function should have at least 1 unit test each (N/A - we have no types, providers, facts or functions)
## Puppet Versions & Features
- [x] Must not rely on experimental Puppet features (like the future parser or in-module hiera data)
- [x] Must be compatible with the Puppet 3 and Puppet Enterprise 3 series.
- [x] Should not directly call out to an ENC like the hiera() function for example.

**Comment by 3flex** [comment]
@jfryman I've refreshed the list and updated with current status.

There are two remaining "musts":
* One in "documentation" section
* One in "metadata" section - close after merging #598
* Candidate version must be >= 1.x

Let me know if you want assistance with anything. I will refresh #451 as well and comment there when I think you can reconsider merging.

I suggest opening a new ticket with a checklist for anything that should be done before 1.x.

**Comment by wyardley** [comment]
I think we now have release > 1.0, but docs is still a mess.
@bastelfreak: do we want to leave this open until we improve the docs? Are we concerned about being #puppetapproved?

**Comment by anarcat** [comment]
we *are* puppet approved now, so i'm just going to close this. there are separate issues (#1245, #451) for improving docs.
